[dialog-report-column-view.cpp] new/delete instead of g_new/malloc

christopherlam opened the pull request:
because the struct has cpp (eg vector) objects which would otherwise leak.

christopherlam committed:
[dialog-report-column-view.cpp] new/delete instead of g_new/g_free

because the struct has vector objects which would otherwise leak.

jralls commented:
Where's the leak? Are `gncp_column_view_edit`s getting destroyed without `gncp_column_view_edit_destroy` being called?

christopherlam commented:
No, gncp_column_view_edit_destroy is being called appropriately but doesn't nullify the string vector

jralls commented:
That seems like a compiler bug if calling `free` doesn't run the default destructor. Did you try calling `available_list.clear()` in `gncp_column_view_edit_destroy` and find that it still leaked something?

christopherlam commented:
I thought `free()` would never call the cpp destructors.

christopherlam committed:
[gnc-engine-guile.cpp] plug runaway GSList* leak

The QofQuery param_list is a GSList of const char*. Every SCM Query
would typically leak the GSList. This change will store the params
into the string cache, and the cached string is passed onto the
QofQuery param.

jralls commented:
Oy, [you're](https://godbolt.org/z/WK7e5rYnf) [right](https://en.cppreference.com/w/cpp/language/delete). That's disappointing because the compiler notices other cases where a variable becomes invalid and arranges for its destruction.
